unlock-app/src/hooks/useLocks.js

```diff
@@ -16,14 +20,48 @@ const config = configure()
  */
 export const useLocks = owner => {
   // Let's retrieve the locks!
-  const [loading, setLoading] = useState(true)
-  const [locks, setLocks] = useState([])
+  const network = useSelector(state => state.network)
   const web3Service = useContext(Web3ServiceContext)
+  const walletService = useContext(WalletServiceContext)
+  const storageService = useContext(StorageServiceContext)
+  const [error, setError] = useState(undefined)
+  const [loading, setLoading] = useState(true)
+
+  // We use a reducer so we can easily add locks as they are retrieved
+  const [locks, addToLocks] = useReducer((locks, lock) => {
+    let l = locks.find(element => element.address === lock.address)
+    if (!l) {
+      // New lock, add it
+      return [lock, ...locks]
+    }
+    // The lock already exists. we merge
+    l = {
+      ...lock,
+      ...l,
```

Comment:
I think here we want something more like

```js
l = {
  ...l,
  ...lock,
}
```

So that the newer data takes precedence